catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/events/EventApplication.java

```diff
@@ -0,0 +1,79 @@
+package org.codice.ddf.catalog.ui.events;
+
+import static spark.Spark.get;
+
+import java.io.PrintWriter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import spark.servlet.SparkApplication;
+
+public class EventApplication implements SparkApplication {
+
+  private static final List<PrintWriter> listeners = new ArrayList<>();
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(EventApplication.class);
+
+  private static final String HEARTBEAT = ": \n\n";
+
+  @Override
+  public void init() {
+    get(
+        "/events",
+        (req, res) -> {
+          try {
+            res.type("text/event-stream; charset=UTF-8");
+            res.header("Connection", "keep-alive");
+            res.header("Cache-Control", "no-cache");
+            res.status(200);
+            PrintWriter out = res.raw().getWriter();
+            synchronized (listeners) {
+              listeners.add(out);
+            }
+            out.write(HEARTBEAT);
+            out.flush();
```

Comment:
are these needed if we write the HEARTBEAT in a loop below? is the flush needed below?

Reply:
fixed